### ISSUE TITLE
关于训练网络中遇到的问题

### ISSUE THREAD

**Issue by Scr-w** [ohhhyeahhh/SiamCAR#48]
作者您好，感谢您开源您的代码供我们学习，我在用这个网络训练了一个其他的数据集，发现训练出来的权重大小和您提供的供测试的权重大小不一样，请问可以帮忙解决一下疑惑吗~非常感谢~期待您的回复

**Comment by Scr-w** [comment]
你好，请问这块else那里的requires_grad参数是不是应该也为True呢，期待您的回复，非常感谢
if current_epoch> = cfg.BACKBONE.TRAIN_EPOCH：
用于cfg.BACKBONE.TRAIN_LAYERS：
用于getattr（ model.backbone，layer）.parameters（）：
param.requires_grad = true
for getattr（model.backbone，layer）.modules（）：
if isinstance（m，nn.BatchNorm2d）：
m.train（）
else：
for model.backbone.parameters（）中的参数：
param.requires_grad = False＃
model.backbone.modules（）中的m值为False ：
if isinstance（m，nn.BatchNorm2d）：
m.eval（）

**Comment by Scr-w** [comment]
你好，请问这块else那里的requires_grad参数是不是应该也为True呢，期待您的回复，非常感谢
if current_epoch >= cfg.BACKBONE.TRAIN_EPOCH:
for layer in cfg.BACKBONE.TRAIN_LAYERS:
for param in getattr(model.backbone, layer).parameters():
param.requires_grad = True
for m in getattr(model.backbone, layer).modules():
if isinstance(m, nn.BatchNorm2d):
m.train()
else:
for param in model.backbone.parameters():
param.requires_grad = False #False
for m in model.backbone.modules():
if isinstance(m, nn.BatchNorm2d):
m.eval()

**Comment by WangJun-ZJUT** [comment]
您好!这段代码的逻辑是在第10轮迭代之前冻结backbone的参数,在10轮迭代之后解冻与后面的head模块一起finetune,关于模型大小是根据model.module.state_dict()与optimizer.state_dict()确定的,在对model和optimizer没有修改的情况下肯定是一致的.

**Comment by Scr-w** [comment]
嗯嗯好的谢谢~我看最新的代码把    trainable_params += [{'params': model.down.parameters(),
                          'lr': cfg.TRAIN.BASE_LR}]这块加上了，请问您后面有没有再加上这个后进一步做实验，实验效果如何呢，期待您的回复~非常感谢

**Comment by Scr-w** [comment]
您好，请问您提供的general_model code: lw7w
got10k_model code: p4zx
LaSOT_model code: 6wer这三个权重中的general_model是指哪些数据训练的权重的，而且general_model 的大小和另外两个权重相差了0.02M，是什么原因呢~非常期待您的回复，非常感谢~

**Comment by Scr-w** [comment]
你好，请问在训练的过程中，前10个epoch和后10个epoch的batch_size需要保持一致吗

**Comment by WangJun-ZJUT** [comment]
您好!general_model使用的coco,det,vid,ytbb四个数据集训练的,关于模型大小这点,模型的参数量是一致的,请参照代码进行试验.

**Comment by Scr-w** [comment]
您好，非常感谢您的回复，请问在训练的过程中，前10个epoch和后10个epoch的batch_size大小需要保持一致吗，我在训练的过程中，为了保证显卡的利用率，就将前10个epoch和后10个epoch的batch_size设置的不一样，然后发现我复现出来的网络性能和论文中差距很大，期待您的回复，感谢


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ohhhyeahhh/SiamCAR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年1月20日(星期三) 下午3:09
收件人:&nbsp;"ohhhyeahhh/SiamCAR"<SiamCAR@noreply.github.com&gt;;
抄送:&nbsp;"豆蔻年华"<512979854@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ohhhyeahhh/SiamCAR] 关于训练网络中遇到的问题 (#48)





 
您好!general_model使用的coco,det,vid,ytbb四个数据集训练的,关于模型大小这点,模型的参数量是一致的,请参照代码进行试验.

**Comment by WangJun-ZJUT** [comment]
您好!batch_size通常是保持一致的

**Comment by Scr-w** [comment]
您好，感谢您的回复，请问论文中的batchsize=32是在已经尝试了更大的batchsize后效果最好的参数设置了吗~期待您的回复，非常感谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ohhhyeahhh/SiamCAR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年1月21日(星期四) 上午10:58
收件人:&nbsp;"ohhhyeahhh/SiamCAR"<SiamCAR@noreply.github.com&gt;;
抄送:&nbsp;"豆蔻年华"<512979854@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ohhhyeahhh/SiamCAR] 关于训练网络中遇到的问题 (#48)





 
您好!batch_size通常是保持一致的

**Comment by WangJun-ZJUT** [comment]
您好,我们没有在batch_size这个参数上搜索过,在数据集足够大的情况下通常batch_size是越大越好的

**Comment by Scr-w** [comment]
好的，非常感谢您的回复，请问如果数据集比较小的话，是因为把batchsize设置的很大，会导致迭代次数变少，从而会影响性能吗，期待您的回复，非常感谢




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ohhhyeahhh/SiamCAR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年1月21日(星期四) 中午12:08
收件人:&nbsp;"ohhhyeahhh/SiamCAR"<SiamCAR@noreply.github.com&gt;;
抄送:&nbsp;"豆蔻年华"<512979854@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ohhhyeahhh/SiamCAR] 关于训练网络中遇到的问题 (#48)





 
您好,我们没有在batch_size这个参数上搜索过,在数据集足够大的情况下通常batch_size是越大越好的

**Comment by WangJun-ZJUT** [comment]
您好,数据集小的情况下需要根据实验情况决定batch_size大小.

**Comment by Scr-w** [comment]
好的，非常感谢


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ohhhyeahhh/SiamCAR"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2021年1月21日(星期四) 中午12:22
收件人:&nbsp;"ohhhyeahhh/SiamCAR"<SiamCAR@noreply.github.com&gt;;
抄送:&nbsp;"豆蔻年华"<512979854@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [ohhhyeahhh/SiamCAR] 关于训练网络中遇到的问题 (#48)





 
您好,数据集小的情况下需要根据实验情况决定batch_size大小.

**Comment by dhy1222** [comment]
> 作者您好，感谢您开源您的代码供我们学习，我在用这个网络训练了一个其他的数据集，发现训练出来的权重大小和您提供的供测试的权重大小不一样，请问可以帮忙解决一下疑惑吗~非常感谢~期待您的回复

您好，请问您解决这个问题了吗？我用作者提供的代码训练的模型只有200多M，但是作者提供的general模型有近400M

**Comment by dhy1222** [comment]
> 作者您好，感谢您开源您的代码供我们学习，我在用这个网络训练了一个其他的数据集，发现训练出来的权重大小和您提供的供测试的权重大小不一样，请问可以帮忙解决一下疑惑吗~非常感谢~期待您的回复

您好，请问您解决这个问题了吗？我用作者提供的代码训练的模型只有200多M，但是作者提供的general模型有近400M。

**Comment by dhy1222** [comment]
> > 作者您好，感谢您开源您的代码供我们学习，我在用这个网络训练了一个其他的数据集，发现训练出来的权重大小和您提供的供测试的权重大小不一样，请问可以帮忙解决一下疑惑吗~非常感谢~期待您的回复
> 
> 您好，请问您解决这个问题了吗？我用作者提供的代码训练的模型只有200多M，但是作者提供的general模型有近400M。

@WangJun-ZJUT 请问您可以帮忙解答一下吗？
